M2/Macaulay2/m2/testing.m2

```diff
@@ -60,9 +60,14 @@ captureTestResult := (desc, teststring, pkg, usermode) -> (
     runString(teststring, pkg, usermode))
 
 loadTestDir := pkg -> (
-    -- TODO: prioritize reading the tests from topSrcdir | "Macaulay2/tests/normal" instead
-    testDir := pkg#"package prefix" |
-        replace("PKG", pkg#"pkgname", currentLayout#"packagetests");
+    testDir := (
+	-- prioritize Core tests from source repository
+	if (pkg === Core and (
+		dir := minimizeFilename(pkg#"source directory" |
+		    "../tests/normal/");
+		isDirectory dir)) then dir
```

Comment:
The correct way to check if we're running from the build directory is to check if `topSrcdir` is not null.

Reply:
That only works if we're running from the *build* directory, though.  The use case that Anton pointed out to me yesterday was when running the regular old system copy of the M2 binary, but using the `--srcdir` command line option to grab Core from the source repository to test a new feature.

Reply:
Why would this even work?

Reply:
That's correct -- it doesn't work a lot of the time exactly for this reason. But it's useful when it does!

Reply:
Is there a solution which involves setting `topSrcdir` in this case as well?

Reply:
`--srcdir` really only adds a directory to the path.  *Maybe* it's the path to Core source, but not necessarily.  So I'm not sure if setting topSrcdir based on its value makes sense.

Reply:
Something about this still deeply bothers me:
```m2
	if (pkg === Core and (
		dir := minimizeFilename(pkg#"source directory" |
		    "../tests/normal/");
```